### PR TITLE
add use cases to developer guide v1.2

### LINKED ISSUE #87: Add use cases to developer guide v1.2

### PR DESCRIPTION
**Pull request by MarcTzh**
Fix #87

### CODE REVIEW

**Comment by GabriellaTeh** [comment]
For use cases for list, add, delete, it should be task not person.

**Comment by BobbyZhouZijian** [comment]
What are left to be done for the use cases? Maybe can add inside description?

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -268,30 +268,101 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
-1.  User requests to list persons
-2.  AddressBook shows a list of persons
-3.  User requests to delete a specific person in the list
-4.  AddressBook deletes the person
+1. User requests to add a specific person to the list.
 
-    Use case ends.
+2. PlaNus adds the person.
+
+   Use case ends.
```
Comment (BobbyZhouZijian) [inline docs/DeveloperGuide.md]: I think there should not be indentation here

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -268,30 +268,101 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+
+
+
+**Use case 02: Delete a task**
+
+**MSS**
+
+1. User requests to list persons.
+
+2. PlaNus shows a list of persons.
```
Comment (BobbyZhouZijian) [inline docs/DeveloperGuide.md]: Should it be "the list of all tasks"?

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -268,30 +268,101 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+
+    Use case resumes at step 1.
+
+
+
+**Use case 02: Delete a task**
+
+**MSS**
+
+1. User requests to list persons.
```
Comment (BobbyZhouZijian) [inline docs/DeveloperGuide.md]: should it be "list all tasks"?

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -268,30 +268,101 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+
+2. PlaNus shows a list of persons.
+
+3. User requests to delete a specific person in the list.
+
+4. PlaNus deletes the person.
+
+   Use case ends.
```
Comment (BobbyZhouZijian) [inline docs/DeveloperGuide.md]: Same as above

File: docs/DeveloperGuide.md [hunk 5]
```diff
@@ -268,30 +268,101 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
 
 **Extensions**
 
-* 2a. The list is empty.
+- 2a. The list is empty.
```
Comment (BobbyZhouZijian) [inline docs/DeveloperGuide.md:310]: are we going to skip displaying or just display an empty list? If we are just displaying an empty list, is it still considered an extension?

File: docs/DeveloperGuide.md [hunk 6]
```diff
@@ -268,30 +268,101 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+
+    Use case resumes at step 2.
+
+    
+
+**Use case 03: List all tasks**
+
+**MSS**
+
+1. User requests to list persons
```
Comment (BobbyZhouZijian) [inline docs/DeveloperGuide.md]: list all tasks

File: docs/DeveloperGuide.md [hunk 7]
```diff
@@ -268,30 +268,101 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+
+2. PlaNus shows a list of persons
```
Comment (BobbyZhouZijian) [inline docs/DeveloperGuide.md]: the list of all tasks

File: docs/DeveloperGuide.md [hunk 8]
```diff
@@ -268,30 +268,101 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+
+   Use case ends.
```
Comment (BobbyZhouZijian) [inline docs/DeveloperGuide.md]: same as above. i'll ignore it for the following parts

File: docs/DeveloperGuide.md [hunk 9]
```diff
@@ -268,30 +268,101 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+
+**Extensions**
+
+- 3a. The given index is invalid.
 
-    * 3a1. AddressBook shows an error message.
+  - 3a1. PlaNus shows an error message.
 
-      Use case resumes at step 2.
+    Use case resumes at step 3.
```
Comment (BobbyZhouZijian) [inline docs/DeveloperGuide.md:373]: Is 'exit' inside the use cases? should we add it?

**Review by BobbyZhouZijian — COMMENTED** [review]
Almost LGTM. Just a few nits to fix.

File: docs/DeveloperGuide.md (hunk 5 above)
Comment (MarcTzh) [reply docs/DeveloperGuide.md:310]: added extension

**Review by GabriellaTeh — APPROVED** [review]
LGTM

**Pull request merged**